dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -139,25 +185,36 @@ def check_pipeline(input_epoch, pipeline_fn, *, devices: list = ['cpu', 'gpu'],
 
     This function conducts qualitative verification. It compares the result of
     running multiple iterations of the same pipeline (with possible varying batch sizes,
-    accoring to `input_epoch`) with results of the ad-hoc created pipelines per iteration
+    according to `input_epoch`) with results of the ad-hoc created pipelines per iteration
 
-    :param input_epoch: List of numpy arrays, where every item is a single batch
+    :param input_epoch: List of input batches, if there is one input a batch can be either
+                        a numpy array or a list, for multiple inputs it can be tuple of lists or
+                        numpy arrays.
     :param pipeline_fn: Function, that returns created (but not built) pipeline.
                         Its signature should be (at least):
                         pipeline_fn(max_batch_size, input_data, device, ...)
     :param devices: Devices to run the check on
     :param eps: Epsilon for mean error
     :param pipeline_fn_args: Additional args to pipeline_fn
     """
+    def get_batch_size(batch):
+        if isinstance(batch, tuple):
+            return get_batch_size(batch[0])
```

Comment:
Is this function the same in `run_pipeline` and `check_pipeline`? If so, maybe extracting to `_get_batch_size`?

Reply:
Done